app/database.py

```diff
@@ -23,83 +25,142 @@ def get_database():
+    conn.close()
+
+
+def get_events():
+    db = get_db()
     cursor = db.cursor()
-    cursor.execute('''SELECT ''' + field + ''' FROM schedules''')
-    all_rows = cursor.fetchall()
-    for row in all_rows:
-        print(row[0])
-
-# initialize the database by running "flask init_database" in
-# the venv.  It will create a new skemaDB.sqlite file in
-# the instance folder
-@click.command('init_database')
-@with_appcontext
-def init_database_command():
-    init_database()
-    event = 'Initialized the database'
-    log.add_event(event)
-    click.echo(event)
+    statement = 'SELECT * FROM winter2017'
+    data = cursor.execute(statement).fetchall()
+    classes = [dict(zip([key[0] for key in cursor.description], row)) for row in data]
+    return str(json.dumps(({'classes': classes})))
 
-# returns any column passed in as an argument.
-# For Example: 'flask fetch_database course_id'
-# will return all of the course ids in the database
-@click.command('fetch_database')
-@click.argument('field')
-@with_appcontext
-def fetch_database_command(field):
-    fetch_database(field)
 
-# fills the database with a file from the given path.
-# Will throw an error if the file passed is not in
-# the correct format.
-@click.command('fill_database')
+def get_profs():
+    db = get_db()
+    cursor = db.cursor()
+    statement = 'SELECT DISTINCT professor FROM winter2017'
+    rows = cursor.execute(statement).fetchall()
+    lst = []
+    for row in rows:
+        lst.append(row[0])
+    return lst
+
+
+def get_subject():
+    db = get_db()
+    cursor = db.cursor()
+    statement = 'SELECT DISTINCT (SUBSTR(title, 0, 5)) FROM winter2017'
+    rows = cursor.execute(statement).fetchall()
+    lst = []
+    for row in rows:
+        lst.append(row[0])
+    return lst
+
+
+'''
+The following methods are for command line manipulation of the this app.  They are not necessary when working within
+the webpage.  
+
+fills the database with a file from the given path.Will throw an error if the file passed is not in the correct format.
+'''
+
+
+@click.command('fill_db')
 @click.argument('file')
+#@click.argument('semester')
 @with_appcontext
-def fill_database_command(file):
-    from . import parser
-    parser.parse(file)
-    event = 'Database filled'
+def fill_db_command(file):
+    fill_db(file)
+    event = 'Database filled with ' + file
     log.add_event(event)
+    click.echo(event)
+
 
-# drops the table from the database, removing everything
-# in it.
-@click.command('destroy_database')
+'''
+selects all classes of a queried category.  Can be refined down to year. 
+call 'Flask show_classes <semester> <course type>'
+eg1 Flask show_classes fall2017 CPSC displays all CPSC courses
+eg2 Flask show_classes winter2017 CPSC2 displays all 2nd year courses
+'''
+
+
+@click.command('get_classes')
+@click.argument('semester')
 @with_appcontext
-def destroy_database_command():
-    destroy_database()
-    event = 'Database destroyed'
+def get_classes_command():
```

Comment:
This method is not needed. It was only needed for debugging.  The method it calls is still required.